using-mobile-extensions/adobe-campaignclassic/adobe-campaignclass-c-api-reference.md

```diff
@@ -0,0 +1,261 @@
+
+### registerDevice
+
+The registerDevice API will register a device with your Campaign Classic registration server. It takes the FCM registration token as a parameter along with a user key that identifies the user such as an email address or login name. It can also be given a Map of custom key value pairs that you want to associate with the registration. Lastly, a boolean value is returned in the callback which signals whether the registration was sucessful.
+
+#### Java
+
+#### Syntax
+
+```java
+public static void registerDevice(final String token, final String userKey, final Map<String, Object> additionalParams, final AdobeCallback<Boolean> callback)
+```
+
+#### Example
+
+```java
+/**
+ * Called if InstanceID token is updated. This may occur if the security of
+ * the previous token had been compromised. Note that this is called when the InstanceID token
+ * is initially generated so this is where you would retrieve the token.
+ */
+@Override
+public void onNewToken(String token) {
+    Log.d(TAG, "Refreshed token: " + token);
+
+    // If you want to send messages to this application instance or
+    // manage this apps subscriptions on the server side, send the
+    // Instance ID token to your app server.
+    if (token != null) {
+				System.out.println("FCM SDK registration token received : " + token);
+                // Create a map of additional paramters
+                Map<String, Object> additionalParams = new HashMap<String, Object>();
+            	additionalParams.put("testNum", 12345);
+            	additionalParams.put("testBool", true);
+            	additionalParams.put("brand", "Adobe");
+                // Send the registration info
+				CampaignClassic.registerDevice(token, "user@gmail.com", 					 			additionalParams,new AdobeCallback<Boolean>() {
+					@Override
+					public void call(final Boolean status) {
+						Log.w(TAG, "Status: " + status;
+					}
+				});
+  	}
+}
+```
+
+{% endtab %}
+
+{% tab title="iOS" %}
+
+Follow [Apple's instructions](https://developer.apple.com/library/archive/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/HandlingRemoteNotifications.html#//apple_ref/doc/uid/TP40008194-CH6-SW1) to get your app ready to handle push notifications. Once you receive the APNS token, you will need to send that token along with the device information to Campaign Classic using the registerDevice API.
+
+### registerDevice
+
+The registerDevice API will register a device with your Campaign Classic registration server. It takes the APNS token as a parameter along with a user key that identifies the user such as an email address or login name. It can also be given a dictionary of custom key value pairs that you want to associate with the registration. Lastly, a boolean value is returned in the callback which signals whether the registration was sucessful.
```

Comment:
- In the second sentence, please change "along with" to "with".
- In the second sentence, please add a comma between "user" and "such as"
- In the final sentence, please change from "Lastly" to "As a result"